core/src/test/scala/sbtorgpolicies/github/GitHubOpsTest.scala

```diff
@@ -0,0 +1,567 @@
+        case (Right(_), Right(gHResult)) =>
+          result.map(_.toSet) shouldBeEq Right(gHResult.result.filter(_.merged_at.nonEmpty).toSet)
+      }
+    }
+    check(property)
+  }
+
+  test(
+    "GithubOps.latestPullRequests should return all merged pull requests with date greater than the selected commit") {
+
+    val property = forAll(genGHResPRListMergedFrom2015, ghResponseCommitListArbitrary.arbitrary) {
+      (prResponse: GHResponse[List[PullRequest]], commitsResponse: GHResponse[List[Commit]]) =>
+        resetMocks()
+
+        val commit = Commit(
+          sha = "sha",
+          message = nonExistingMessage,
+          date = dateTimeFormat.print(date2015),
+          url = "http://github.com",
+          login = None,
+          avatar_url = None,
+          author_url = None)
+
+        val newCommitsResponse = commitsResponse map { ghResult =>
+          ghResult.copy(result = commit :: ghResult.result)
+        }
+
+        when(
+          ghRepos.listCommits(
+            any[String],
+            any[String],
+            any[Option[String]],
+            any[Option[String]],
+            any[Option[String]],
+            any[Option[String]],
+            any[Option[String]],
+            any[Option[Pagination]]))
+          .thenReturn(Free.pure[GitHub4s, GHResponse[List[Commit]]](newCommitsResponse))
+
+        when(ghPullRequests.list(any[String], any[String], any[List[PRFilter]]))
+          .thenReturn(Free.pure[GitHub4s, GHResponse[List[PullRequest]]](prResponse))
+
+        val result: Either[GitHubException, List[PullRequest]] =
+          githubOps.latestPullRequests(branch, "", nonExistingMessage)
+
+        (commitsResponse, prResponse) match {
+          case (Left(e), _) =>
+            result shouldBeEq Left(GitHubException(s"GitHub returned an error: ${e.getMessage}", Some(e)))
+          case (_, Left(e)) =>
+            result shouldBeEq Left(GitHubException(s"GitHub returned an error: ${e.getMessage}", Some(e)))
+          case (Right(_), Right(gHResult)) =>
+            val expected =
+              gHResult.result.filter(_.merged_at.exists(s => dateTimeFormat.parseDateTime(s).isAfter(date2015))).toSet
+            result.map(_.toSet) shouldBeEq Right(expected)
+        }
+    }
+    check(property)
+  }
+
+  test("GithubOps.fetchHeadCommit works as expected") {
+    val property = forAll { refResponse: GHResponse[NonEmptyList[Ref]] =>
+      resetMocks()
+
+      import syntax._
+
+      when(ghGitData.getReference(any[String], any[String], any[String]))
+        .thenReturn(Free.pure[GitHub4s, GHResponse[NonEmptyList[Ref]]](refResponse))
+
+      val result: Either[GitHubException, Ref] = githubOps.fetchHeadCommit(branch).execE
+
+      verify(ghGitData).getReference(owner, repo, ref)
+
+      refResponse match {
+        case Left(e) =>
+          result shouldBeEq Left(GitHubException(s"GitHub returned an error: ${e.getMessage}", Some(e)))
```

Comment:
Seems like `Left(GitHubException(s"GitHub returned an error: ${e.getMessage}", Some(e))` is being repeated multiple times in the file, does it make sense extract it up to one class val?